barretenberg/cpp/src/barretenberg/stdlib/recursion/honk/verifier/protogalaxy_recursive_verifier.cpp

```diff
@@ -0,0 +1,305 @@
+        transcript->template receive_from_prover<uint32_t>(domain_separator + "_pub_inputs_offset");
+
+    inst->pub_inputs_offset = uint32_t(pub_inputs_offset.get_value());
+
+    auto labels = inst->commitment_labels;
+    auto& witness_commitments = inst->witness_commitments;
+    witness_commitments.w_l = transcript->template receive_from_prover<Commitment>(domain_separator + "_" + labels.w_l);
+    witness_commitments.w_r = transcript->template receive_from_prover<Commitment>(domain_separator + "_" + labels.w_r);
+    witness_commitments.w_o = transcript->template receive_from_prover<Commitment>(domain_separator + "_" + labels.w_o);
+
+    auto eta = transcript->get_challenge(domain_separator + "_eta");
+    witness_commitments.sorted_accum =
+        transcript->template receive_from_prover<Commitment>(domain_separator + "_" + labels.sorted_accum);
+    witness_commitments.w_4 = transcript->template receive_from_prover<Commitment>(domain_separator + "_" + labels.w_4);
+
+    auto [beta, gamma] = transcript->get_challenges(domain_separator + "_beta", domain_separator + "_gamma");
+    witness_commitments.z_perm =
+        transcript->template receive_from_prover<Commitment>(domain_separator + "_" + labels.z_perm);
+    witness_commitments.z_lookup =
+        transcript->template receive_from_prover<Commitment>(domain_separator + "_" + labels.z_lookup);
+
+    const FF public_input_delta = bb::honk::compute_public_input_delta<Flavor>(
+        inst->public_inputs, beta, gamma, inst->instance_size, inst->pub_inputs_offset);
+    const FF lookup_grand_product_delta =
+        bb::honk::compute_lookup_grand_product_delta<FF>(beta, gamma, inst->instance_size);
+    inst->relation_parameters =
+        RelationParameters<FF>{ eta, beta, gamma, public_input_delta, lookup_grand_product_delta };
+
+    for (size_t idx = 0; idx < NUM_SUBRELATIONS - 1; idx++) {
+        inst->alphas[idx] = transcript->get_challenge(domain_separator + "_alpha_" + std::to_string(idx));
+    }
+
+    inst->verification_key = std::make_shared<VerificationKey>(inst->instance_size, inst->public_input_size);
+    auto vk_view = inst->verification_key->get_all();
+    auto vk_labels = labels.get_precomputed();
+    for (size_t idx = 0; idx < vk_labels.size(); idx++) {
+        vk_view[idx] = transcript->template receive_from_prover<Commitment>(domain_separator + "_" + vk_labels[idx]);
+    }
+}
+
+// TODO(https://github.com/AztecProtocol/barretenberg/issues/795): The rounds prior to actual verifying are common
+// between decider and folding verifier and could be somehow shared so we do not duplicate code so much.
+template <class VerifierInstances> void ProtoGalaxyRecursiveVerifier_<VerifierInstances>::prepare_for_folding()
+{
+    auto index = 0;
+    auto inst = instances[0];
+    auto domain_separator = std::to_string(index);
+    const auto is_accumulator = transcript->template receive_from_prover<bool>(domain_separator + "is_accumulator");
```

Comment:
Interesting - I suppose we could use a similar flag in the merge protocol to determine whether or not we have a previous transcript to merge..